### PR TITLE
finish code challenge

### PR DESCRIPTION
**Pull request by tylersayvetz**

### CODE REVIEW

File: code-challenges/quick-sort/quick-sort.test.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  if (leftLimit >= rightLimit) return array;
+
+  //first partition the array, returning the pivotIndex (the index of the pivot after the rough sort)
+  const pivotIndex = partition(array, leftLimit, rightLimit);
+
+  //then make a recursive call to the left
+  quickSort(array, leftLimit, pivotIndex - 1)
+
+  //then make a recursive call to the right
+  quickSort(array, pivotIndex, rightLimit)
```
Comment (XiaojuanCai) [inline code-challenges/quick-sort/quick-sort.test.js]: Do we need to include pivotIndex here? The entry at pivotIndex is already in the correct place.

File: code-challenges/quick-sort/quick-sort.test.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  for (let i = leftLimit; i < rightLimit; i++) {
+
+    //!if the value at i is less than the pivot, swap the value at pIndex and the value at i.
+    // [8,4,23,41,13,15]   pivot is 15. 
+    //so we would swap 8 with itself on the first iteration and increment both pindex and i.
+    //on the third iteration we 23 is not less than 15, so only i is incremented. pindex wont be incremented until the 5th iteration.
+    //on the fifth iteration 13 is less than 15, so 13 and 23 switch places. Both pindex and i are incremented. 
+    // so now: [8,4,13,41,23,15]
+    if (array[i] <= pivotValue) {
+      swap(array, i, pIndex);
```
Comment (XiaojuanCai) [inline code-challenges/quick-sort/quick-sort.test.js]: What about making it even more efficient: when i === pIndex, we will not swap, only increase pIndex.

File: code-challenges/quick-sort/quick-sort.test.js (hunk 1 above)
Comment (tylersayvetz) [reply code-challenges/quick-sort/quick-sort.test.js]: great catch !

File: code-challenges/quick-sort/quick-sort.test.js (hunk 2 above)
Comment (tylersayvetz) [reply code-challenges/quick-sort/quick-sort.test.js]: another perfect observation! thank you

**Pull request merged**